plugin.py

```diff
@@ -16,3 +20,23 @@ class LspTypescriptPlugin(NpmClientHandler):
     server_binary_path = os.path.join(
         server_directory, 'node_modules', 'typescript-language-server', 'lib', 'cli.js'
     )
+
+    def on_ready(self, api) -> None:
+        api.on_request('_typescript.rename', self._on_typescript_rename)
+
+    def _on_typescript_rename(self, textDocumentPositionParams, respond):
+        view = sublime.active_window().open_file(
+            uri_to_filename(textDocumentPositionParams['textDocument']['uri'])
+        )
+
+        if view:
+            lsp_point = Point.from_lsp(textDocumentPositionParams['position'])
+            point = point_to_offset(lsp_point, view)
+
+            sel = view.sel()
+            sel.clear()
+            sel.add_all([point])
+            view.run_command('lsp_symbol_rename')
+
+        # Server doesn't require any specific response.
+        respond(None)
```

Comment:
I wounder if this works, because I am not sure if the server sends a response_id?

Reply:
It does send it as it's required by the protocol. This `apiWrapper` abstraction I've created just hides it away.

Reply:
I mean, it sends the `request_id`, not `response_id`. We use that as our `response_id` when replying.